Tests/SwiftParserTest/DeclarationTests.swift

```diff
@@ -3473,6 +3473,28 @@ final class DeclarationTests: ParserTestCase {
     )
   }
 
+  func testBorrowAndMutateAccessors() {
+    assertParse(
+      """
+      public class Klass {}
+
+      public struct Wrapper {
+        var _otherK: Klass
+
+        var k1: Klass {
+          borrow {
```

Comment:
I just want to mention that this will be source breaking for the following code, which is currently valid. I’ll leave it up to you to decide whether that source breakage is acceptable.

```swift
func borrow<T>(_ x: () -> T) -> T {
    return x()
}

var x: Int {
    borrow {
        1
    }
}
```

Reply:
Thanks for pointing that out. This is currently experimental and is not an immediate problem. We may gate the feature on language version in the future if source breakage becomes a concern.